src/som/interpreter/actors/ResolvePromiseNode.java

```diff
@@ -6,32 +6,56 @@
 import com.oracle.truffle.api.instrumentation.Instrumentable;
 import com.oracle.truffle.api.source.SourceSection;
 
+import som.VM;
 import som.interpreter.actors.SPromise.SResolver;
-import som.interpreter.nodes.nary.BinaryComplexOperation;
+import som.interpreter.nodes.nary.QuaternaryExpressionNode;
+import som.interpreter.nodes.nary.UnaryExpressionNode;
 import som.primitives.Primitive;
 
 
 @GenerateNodeFactory
-@Primitive(primitive = "actorsResolve:with:")
+@Primitive(primitive = "actorsResolve:with:isBPSender:isBPRcvr:")
```

Comment:
think the names also need to be adapted to new terminology (also in the Smalltalk code)